manim/__main__.py

```diff
@@ -1,14 +1,168 @@
+                    commands.append("open")
+
+                if file_writer_config["show_file_in_finder"]:
+                    commands.append("-R")
+
+                commands.append(file_path)
+
+                # commands.append("-g")
+                FNULL = open(os.devnull, 'w')
+                sp.call(commands, stdout=FNULL, stderr=sp.STDOUT)
+                FNULL.close()
+
+    if file_writer_config["quiet"]:
+        sys.stdout.close()
+        sys.stdout = curr_stdout
+
+
+def is_child_scene(obj, module):
+    return (inspect.isclass(obj)
+            and issubclass(obj, Scene)
+            and obj != Scene
+            and obj.__module__.startswith(module.__name__))
+
+
+def prompt_user_for_choice(scene_classes):
+    num_to_class = {}
+    for count, scene_class in enumerate(scene_classes):
+        count += 1  # start with 1 instead of 0
+        name = scene_class.__name__
+        print("%d: %s" % (count, name))
+        num_to_class[count] = scene_class
+    try:
+        import re
+        user_input = input(constants.CHOOSE_NUMBER_MESSAGE)
+        return [num_to_class[int(num_str)]
+                for num_str in re.split(r"\s*,\s*", user_input.strip())]
+    except KeyError:
+        logger.error(constants.INVALID_NUMBER_MESSAGE)
+        sys.exit(2)
+    except EOFError:
+        sys.exit(1)
+
+
+def get_scenes_to_render(scene_classes):
+    if not scene_classes:
+        logger.error(constants.NO_SCENE_MESSAGE)
+        return []
+    if file_writer_config["write_all"]:
+        return scene_classes
+    result = []
+    for scene_name in file_writer_config["scene_names"]:
+        found = False
+        for scene_class in scene_classes:
+            if scene_class.__name__ == scene_name:
+                result.append(scene_class)
+                found = True
+                break
+        if not found and (scene_name != ""):
+            logger.error(
+                constants.SCENE_NOT_FOUND_MESSAGE.format(
+                    scene_name
+                )
+            )
+    if result:
+        return result
+    return [scene_classes[0]] if len(scene_classes) == 1 else prompt_user_for_choice(scene_classes)
+
+
+def get_scene_classes_from_module(module):
+    if hasattr(module, "SCENES_IN_ORDER"):
+        return module.SCENES_IN_ORDER
+    else:
+        return [
+            member[1]
+            for member in inspect.getmembers(
+                module,
+                lambda x: is_child_scene(x, module)
+            )
+        ]
+
+
+def get_module(file_name):
+    if file_name == "-":
+        module = types.ModuleType("input_scenes")
+        code = sys.stdin.read()
+        try:
+            exec(code, module.__dict__)
+            return module
+        except Exception as e:
+            logger.error(f"Failed to render scene: {str(e)}")
+            sys.exit(2)
+    else:
+        if os.path.exists(file_name):
+            module_name = file_name.replace(os.sep, ".").replace(".py", "")
```

Comment:
Could have multiple `.py`s in its name, so make sure the last one(s) is(are) the one(s) we're replacing
Also `import re` at the top level (we had alread imported `re` locally somewhere else i think so just move that to the top level if so)
```suggestion
            module_name = re.replace(r"(?:\.py)+$", "", file_name.replace(os.sep, "."))
```

Reply:
The non-capturing group complicates the regex, and I don't think it has any benefit. Additionally, this doesn't handle cases like file.py.txt.py so it may be better to just take the section of file_name up to the first ..

Reply:
The non-capturing group complicates the regex, and I don't think it has any benefit. Additionally, this doesn't handle cases like `file.py.txt.py` so it may be better to just take the section of `file_name` up to the first `.`.

Reply:
doesn't seem complicated at all to me, but your other point is reasonable
```suggestion
            module_name = re.replace(r"\..+$", "", file_name.replace(os.sep, "."))
```

Reply:
I'm going to add this, but honestly I think that explicitly discouraging file names with two `.py` substrings in the name is better than adding this complexity. (Yes, it is not out-of-this-world complex, but it seems unnecessary to support such an edge case.)

Reply:
meh, you never know i guess

Reply:
Also, `re.replace` doesn't exist. Did you mean `re.sub`?